src/hardwareLayer/io/SignalGenerator.cpp

```diff
@@ -100,8 +110,32 @@ Signal SignalGenerator::nextSignal() {
 	return signal;
 }
 
+bool SignalGenerator::dealWithClatter(Signal signal){
+	auto timeSinceLastInterrupt = std::chrono::duration_cast <std::chrono::milliseconds> (std::chrono::steady_clock::now() - time);
+	time = std::chrono::system_clock::now();
```

Comment:
ich würde die `time` variable z.B. `lastClatterTime` nennen. 

die erste zeile der Funktion wäre dann so etwas:

`std::chrono::steady_clock::time_point currentClatterTime = std::chrono::steady_clock::now()`

und die letzte `lastClatterTime = currentClatterTime`

Dann sparst du dir den zweimaligen zeitaufwändigeren `std::chrono::steady_clock::now()` Aufruf und die lesbarkeit wird etwas eindeutiger.

Reply:
theoretisch würde auch das in der ersten Zeile möglich sein: (um um die temporäre variable herum zu kommen..)
`auto timeSinceLastInterrupt = std::chrono::duration_cast <std::chrono::milliseconds> (time = std::chrono::steady_clock::now() - time);`
aber so etwas geht heutzutage denk ihc nicht mehr ;)